kb/2025-03-20/understanding_Harvester_storage_consumption_during_VM_migration.md

```diff
@@ -0,0 +1,65 @@
+---
+title: Understanding Harvester storage consumption when migrating a VM
+description: 'This article explains how Harvester allocates storage during virtual machine (VM) migration using the VM import controller, focusing on the storage implications when migrating larger VMs. Understanding these mechanics is crucial for effective storage planning and ensuring a smooth migration process.'
+slug: understanding_harvester_storage_consumption_during_VM_migration
+authors:
+  - name: Devendra Kulkarni
+    title: Customer Support Engineer IV
+    url: https://github.com/devendra-suse
+    image_url: https://github.com/devendra-suse.png
+tags: [virt-import-controller, disk-usage, longhorn]
+hide_table_of_contents: false
+---
+
+This article explains how Harvester allocates storage during virtual machine (VM) migration using the VM import controller, focusing on the storage implications when migrating larger VMs. Understanding these mechanics is crucial for effective storage planning and ensuring a smooth migration process.
+
+## Harvester's Storage Allocation Mechanism:
+
+Harvester employs a distributed block-storage architecture utilizing Longhorn. Key concepts to understand include:
+
+- **Backing Images:** These are the base disk images of your VMs, stored as qcow2 files.
+- **Volume Replicas:** Harvester creates multiple replicas of the VM's volumes across cluster nodes for redundancy and high availability. By default, three copies of the backing image and three volume replicas are created.
+- **Sparse Files:** Harvester uses linux sparse files, meaning that storage space is allocated but not immediately consumed. Physical disk space is only utilized as data is written to the VM.
+- **Snapshots:** Both system and user-created snapshots contribute to storage consumption.
+    * **System Snapshots:** Used for internal operations, like replica rebuilding.
+    * **User Snapshots:** Created by users for data protection.
+- **Volume Head:** The active writable layer of the volume.
+
+
+## Storage Consumption Breakdown:
+
+Each volume replica consists of:
+
+- The backing image
+- Any snapshots (system or user)
+- The volume head
+
+
+## Storage Implications for a large size VM:
+
+To illustrate storage consumption, let's consider migrating a 500GB VM.
+
+- **Default Configuration:** Harvester's default configuration involves three copies of the backing image and three volume replicas.
+- **Worst-Case Scenario:** In the worst-case scenario, we assume all allocated space is utilized. This means each node will store:
+    - One copy of the 500GB backing image.
+    - One 500GB volume head.
+    - Potential space for snapshots, max 500GB
```

Comment:
All other enumerations end with a period.
```suggestion
    - Potential space for snapshots, max 500GB.
```

Reply:
Resolved, modified to `maximum 500G.`